packages/bff/src/routers/genericRouter.ts

```diff
@@ -6,17 +6,44 @@ import {
   zodiosValidationErrorToApiProblem,
 } from "pagopa-interop-commons";
 import { bffApi } from "pagopa-interop-api-clients";
+import { toolsServiceBuilder } from "../services/toolsService.js";
+import { PagoPAInteropBeClients } from "../providers/clientProvider.js";
+import { fromBffAppContext } from "../utilities/context.js";
+import { emptyErrorMapper } from "../utilities/errorMappers.js";
+import { makeApiProblem } from "../model/domain/errors.js";
 
 const toolsRouter = (
-  ctx: ZodiosContext
+  ctx: ZodiosContext,
+  clients: PagoPAInteropBeClients
 ): ZodiosRouter<ZodiosEndpointDefinitions, ExpressContext> => {
   const toolsRouter = ctx.router(bffApi.toolsApi.api, {
     validationErrorHandler: zodiosValidationErrorToApiProblem,
   });
 
-  toolsRouter.post("/tools/validateTokenGeneration", async (_req, res) =>
-    res.status(501).send()
-  );
+  const toolsService = toolsServiceBuilder(clients);
+
+  toolsRouter.post("/tools/validateTokenGeneration", async (req, res) => {
+    const ctx = fromBffAppContext(req.ctx, req.headers);
+
+    try {
+      const result = await toolsService.validateTokenGeneration(
+        req.body.client_id,
+        req.body.client_assertion,
+        req.body.client_assertion_type,
+        req.body.grant_type,
+        ctx
+      );
+      return res.status(200).json(result).end();
```

Comment:
Port here the same approach Sandro is implementing in #1001 so that we have the parsing also on this new route